Cabal/src/Distribution/Fields/Pretty.hs

```diff
@@ -35,6 +35,7 @@ import qualified Text.PrettyPrint as PP
 data PrettyField ann
     = PrettyField ann FieldName PP.Doc
     | PrettySection ann FieldName [PP.Doc] [PrettyField ann]
+    | PrettyEmpty
```

Comment:
Why this change?

Reply:
I'd suggesting trying whether using `Maybe (PrettyField ann)` and collecting `[Maybe (PrettyField ann)]` and `catMaybes` that into sections.

Another option is to have

```haskell
    | PrettyEmpty ann
```

so 

```haskell
prettyFieldAnn :: PrettyField ann -> ann
```

is definable. I.e. every constructor of `PrettyField` can be annotated.

(There is such function for parser `Field`: https://hackage.haskell.org/package/Cabal-3.4.0.0/docs/Distribution-Fields-Field.html#v:fieldAnn)

Reply:
I like the `PrettyEmpty ann` suggestion. The use of `Maybe` here is what i'm trying to avoid, since we must accept `Maybe` returns and the use of `catMaybes` in every bit of code that interfaces with the formatter, when it's such a small and useful addition to just add directly in the printer.

Reply:
it occurred to me that we can still write `prettyFieldAnn`, we just have to choose a sensible annotation default. That is, we would not have `PrettyField ann -> ann`, but rather 

```haskell
prettyFieldAnn :: Default ann => PrettyField ann -> ann
```

Or `Monoid ann` which could provide a mempty in the case of the printer. Just adding that case for `PrettyEmpty` traded in 20 lines of code in 3 modules for 3 lines of code in 1 module, and I have less finicky data prep to be concerned with in the stanza creation. I'm inclined to keep it as-is.